examples/examples/data-channels-flow-control/data-channels-flow-control.rs

```diff
@@ -1,368 +1,263 @@
+};
 
 const BUFFERED_AMOUNT_LOW_THRESHOLD: usize = 512 * 1024; // 512 KB
 const MAX_BUFFERED_AMOUNT: usize = 1024 * 1024; // 1 MB
 
-async fn set_remote_description(pc: &Arc<RTCPeerConnection>, sdp_str: &str) -> Result<()> {
-    let desc = serde_json::from_str::<RTCSessionDescription>(sdp_str)?;
-
-    // Apply the desc as the remote description
-    pc.set_remote_description(desc).await?;
-
-    Ok(())
-}
-
-async fn create_offerer() -> Result<Arc<RTCPeerConnection>> {
-    // Create a MediaEngine object to configure the supported codec
-    let mut m = MediaEngine::default();
+async fn create_peer_connection() -> RTCPeerConnection {
+    let mut media_engine = MediaEngine::default();
 
-    // Register default codecs
-    m.register_default_codecs()?;
+    media_engine.register_default_codecs().unwrap();
 
-    // Create a InterceptorRegistry. This is the user configurable RTP/RTCP Pipeline.
-    // This provides NACKs, RTCP Reports and other features. If you use `webrtc.NewPeerConnection`
-    // this is enabled by default. If you are manually managing You MUST create a InterceptorRegistry
-    // for each PeerConnection.
-    let mut registry = Registry::new();
+    let mut interceptor_registry = Registry::new();
 
-    // Use the default set of Interceptors
-    registry = register_default_interceptors(registry, &mut m)?;
+    interceptor_registry =
+        register_default_interceptors(interceptor_registry, &mut media_engine).unwrap();
 
-    // Create the API object with the MediaEngine
     let api = APIBuilder::new()
-        .with_media_engine(m)
-        .with_interceptor_registry(registry)
+        .with_media_engine(media_engine)
+        .with_interceptor_registry(interceptor_registry)
         .build();
 
-    // Prepare the configuration
+    let ice_servers = vec![RTCIceServer {
+        ..Default::default()
+    }];
+
     let config = RTCConfiguration {
-        ice_servers: vec![RTCIceServer {
-            //urls: vec!["stun:stun.l.google.com:19302".to_owned()],
-            ..Default::default()
-        }],
+        ice_servers,
         ..Default::default()
     };
 
-    // Create a new RTCPeerConnection
-    let pc = Arc::new(api.new_peer_connection(config).await?);
+    api.new_peer_connection(config).await.unwrap()
+}
 
-    let ordered = Some(false);
-    let max_retransmits = Some(0u16);
+async fn create_requester() -> RTCPeerConnection {
+    let pc = create_peer_connection().await;
 
     let options = Some(RTCDataChannelInit {
-        ordered,
-        max_retransmits,
+        ordered: Some(false),
+        max_retransmits: Some(0u16),
         ..Default::default()
     });
 
-    let (send_more_ch_tx, mut send_more_ch_rx) = tokio::sync::mpsc::channel::<()>(1);
-    let send_more_ch_tx = Arc::new(send_more_ch_tx);
+    let (more_can_be_sent, mut maybe_more_can_be_sent) = tokio::sync::mpsc::channel(1);
+    let dc = pc.create_data_channel("data", options).await.unwrap();
+
+    let shared_dc = dc.clone();
```

Comment:
A neat trick when capturing values in closures/async blocks is to create a new block and shadow the captured variables with the same name, i.e.,


```rust
{
    let dc = dc.clone();
    dc.on_open(...
}
```